src/Idris/Parser.hs

```diff
@@ -595,7 +595,7 @@ fnDecl' syn = checkDeclFixity $
                  return (PTy doc argDocs syn fc opts' n nfc ty)
             <|> postulate syn
             <|> caf syn
-            <|> pattern syn
+            <|> pattern syn)
```

Comment:
Why did you add parantheses here?

Reply:
In Megaparsec, `<?>` has the same precedence as `$` and doesn't associate, so this became an error without the parentheses.